spec.md

```diff
@@ -164,48 +164,52 @@ capability.
 ### Volume Lifecycle
 
 ```
-   CreateVolume +------------+ DeleteVolume
- +------------->|  CREATED   +--------------+
- |              +---+----^---+              |
- |       Controller |    | Controller       v
-+++         Publish |    | Unpublish       +++
-|X|          Volume |    | Volume          | |
-+-+             +---v----+---+             +-+
-                | NODE_READY |
-                +---+----^---+
-               Node |    | Node
-            Publish |    | Unpublish
-             Volume |    | Volume
-                +---v----+---+
-                | PUBLISHED  |
-                +------------+
+   CreateVolume +------------+          DeleteVolume
+ +------------->|  CREATED   +------------------------+
+ |              |            +<-----------+           |
+ |              +---+----^---+            |           |
+ |       Controller |    | Controller     |           v
++++         Publish |    | Unpublish      |          +++
+|X|          Volume |    | Volume         |          | |
++-+             +---v----+---+            |          +-+
+                | NODE_READY |            |
+                +---+----^---+            | Node
+               Node |    | Node           | Unpublish
+            Publish |    | Unpublish      | Volume
+             Volume |    | Volume         | (forced)
+                +---v----+---+       +----+----------+
+                | PUBLISHED  +------>| QUARANTINED_P |
+                +------------+       +---------------+
+                   ControllerUnpublishVolume(fenced)
```

Comment:
What if node is shutdown in the state of `NODE_READY` ? only in state of `PUBLISHED` , state can be moved to `QUARANTINED_P ` ?

Reply:
Right, the QUARANTINE_* states are only designed to cover the case where ControllerUnpublishVolume() is used while in a state where it would currently be forbidden, which include PUBLISHED and VOL_READY. If the node is shut down/rebooted, that's a separate case to consider, but without calling ControllerUnpublishVolume(), it should not affect anything.

Reply:
Thank you for your reply. You mean when the status is not only `PUBLISHED` but also `VOL_READY` , ControllerUnpublishVolume() can be used? If so, in case of bellow figure, arrow to the `QUARANTINED_S` should be come from `VOL_READY` also? I mean, bellow operation will also be allowed?
```
               ControllerUnpublishVolume(fenced)
VOL_READY -----------------------------------------> QUARANTINED_SP 
```

Reply:
Yeah there should be another arrow in the diagram but it will be very hard to squeeze it into the ASCII art. I will try.

Reply:
@bswartz 

How about using a tool like [this](https://mermaid-js.github.io/mermaid-live-editor/)? 
Just several human readable lines like below will generates an equivalent output to the ASCII art.

```
stateDiagram-v2
    [*] -->  CREATED : CreateVolume
    CREATED --> [*] : DeleteVolume
    CREATED --> NODE_READY : Controller Publish Volume
    NODE_READY --> CREATED : Controller Unpublish Volume
    NODE_READY --> VOL_READY : Node Stage Volume
    VOL_READY --> NODE_READY : Node Unstage Volume
    VOL_READY --> PUBLISHED : Node Publish Volume
    VOL_READY --> QUARANTINED_S : Controller Unpublish Volume (fenced)
    PUBLISHED --> VOL_READY : Node Unpublish Volume
    PUBLISHED --> QUARANTINED_SP : Controller Unpublish Volume (fenced)
    QUARANTINED_SP --> QUARANTINED_S : Node Unpublish Volume(forced)
    QUARANTINED_S --> CREATED : Node Unstage Volume (forced)
```

Reply:
Completely agree that that's a better way, but changing the format of these diagrams is beyond the scope of my proposal. Maybe submit a separate PR that replaces the ASCII art with more readable figures? Assuming that other PR merges before this, I'd be happy to rebase on top of it and greatly simplify my changes to the state diagram.

Reply:
Agree. I don't mean to block this PR to be merged unless you use the tool. You may only need to consider recreating the diagram with the tool, if you ever need another big change.

Reply:
@bswartz thank you for your reply. And also, I understand that modifying the diagram is hard.